File: 0797-all-paths-from-source-to-target/0797-all-paths-from-source-to-target.py
class Solution:
    def allPathsSourceTarget(self, graph: List[List[int]]) -> List[List[int]]:
        ans = []        
        def dfs(tmpPath,src,end):
            if src == end:
                ans.append(tmpPath)
            for nxt in graph[src]:
                dfs(tmpPath +[nxt],nxt,end)
            
        dfs([0],0,len(graph) - 1)
        return ans

File: 0797-all-paths-from-source-to-target/0797-all-paths-from-source-to-target.cpp
class Solution {

    vector<vector<int>> ans;

public:
    vector<vector<int>> allPathsSourceTarget(vector<vector<int>>& graph) {
        vector<int> tmpPath;
        dfs(graph,tmpPath,0,graph.size()-1);

        return ans;
        
        
    }
    void dfs(vector<vector<int>>& graph, vector<int> &tmpPath,int src, int end){      
        tmpPath.push_back(src);
        if(src == end){
            ans.push_back(tmpPath);
        }
        
        for(int x: graph[src]){
            dfs(graph,tmpPath,x,graph.size()-1);
        }

        
        tmpPath.pop_back();
        
        
    } 
        
        
    
    
};